src/main/java/org/jenkins/tools/test/maven/ExternalMavenRunner.java

```diff
@@ -28,41 +28,50 @@ public class ExternalMavenRunner implements MavenRunner {
 
     private static final Logger LOGGER = Logger.getLogger(ExternalMavenRunner.class.getName());
 
-    private static final String DISABLE_DOWNLOAD_LOGS = "-ntp";
+    @CheckForNull private final File externalMaven;
 
-    @CheckForNull private File mvn;
+    @CheckForNull private final File m2Settings;
+
+    @NonNull private final List<String> mavenArgs;
 
     /**
      * Constructor.
      *
-     * @param mvn Path to Maven. If {@code null}, a default Maven executable from {@code PATH} will
-     *     be used
+     * @param externalMaven Path to Maven. If {@code null}, a default Maven executable from {@code
+     *     PATH} will be used
      */
-    public ExternalMavenRunner(@CheckForNull File mvn) {
-        this.mvn = mvn;
+    public ExternalMavenRunner(
+            @CheckForNull File externalMaven,
+            @CheckForNull File m2Settings,
+            @NonNull List<String> mavenArgs) {
+        this.externalMaven = externalMaven;
+        this.m2Settings = m2Settings;
+        this.mavenArgs = mavenArgs;
     }
 
     @Override
-    public void run(Config config, File baseDirectory, File buildLogFile, String... goals)
+    public void run(
+            Map<String, String> properties, File baseDirectory, File buildLogFile, String... args)
```

Comment:
I renamed the varargs parameter from `goals` to `args` because it isn't always strictly called with goals. That said, I tried to make sure that whenever it was called with _properties_ that those were moved into the first argument instead for consistency.